adapters/AppLovin/AppLovinAdapter/GADMAdapterAppLovinNative.m

```diff
@@ -0,0 +1,159 @@
+//
+//  GADMAdapterAppLovinNative.m
+//  SDK Network Adapters Test App
+//
+//  Created by Santosh Bagadi on 4/11/18.
+//  Copyright © 2018 AppLovin Corp. All rights reserved.
+//
+
+#import "GADMAdapterAppLovinNative.h"
+
+#import <AppLovinSDK/AppLovinSDK.h>
+#import "GADMAdapterAppLovin.h"
+#import "GADMAdapterAppLovinConstant.h"
+#import "GADMAdapterAppLovinUtils.h"
+#import "GADMAdapterAppLovinExtras.h"
+#import "GADMAdapterAppLovinQueue.h"
+#import "GADMAppLovinMediatedNativeUnifiedAd.h"
+#import "GADMAppLovinMediatedNativeAppInstallAd.h"
+
+@interface GADMAdapterAppLovinNative () <ALNativeAdLoadDelegate, ALNativeAdPrecacheDelegate>
+
+@property(nonatomic, weak) id<GADMAdNetworkConnector> connector;
+@property(nonatomic, strong) ALSdk *sdk;
+@property(nonatomic, strong) NSArray *adTypes;
+
+@end
+
+@implementation GADMAdapterAppLovinNative
+
++ (NSString *)adapterVersion {
+  return GADMAdapterAppLovinConstant.adapterVersion;
+}
+
+- (instancetype)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)connector {
+  self = [super init];
+  if (self) {
+    self.connector = connector;
+    self.sdk = [GADMAdapterAppLovinUtils retrieveSDKFromCredentials:connector.credentials];
+
+    if (!self.sdk) {
+      [GADMAdapterAppLovinUtils log:@"Failed to initialize SDK"];
+    }
+  }
+  return self;
+}
+
+- (void)getBannerWithSize:(GADAdSize)adSize {
+  NSLog(@"Incorrect class called for banner request. Use GADMAdapterAppLovin for banner ad"
+        @" requeests.");
+  NSError *error = [NSError errorWithDomain:GADMAdapterAppLovinConstant.errorDomain
+                                       code:kGADErrorInvalidRequest userInfo:nil];
+  [self.connector adapter:self didFailAd:error];
+}
+
+- (void)getInterstitial {
+  NSLog(@"Incorrect class called for banner request. Use GADMAdapterAppLovin for interstitial ad"
+        @" requeests.");
+  NSError *error = [NSError errorWithDomain:GADMAdapterAppLovinConstant.errorDomain
+                                       code:kGADErrorInvalidRequest userInfo:nil];
+  [self.connector adapter:self didFailAd:error];
+}
+
+- (BOOL)isBannerAnimationOK:(GADMBannerAnimationType)animType {
+  return YES;
```

Comment:
I know this is probably never going to be called and it's a bit nitpicky, but for anything that's "off" shouldn't `NO` make more sense? Like if you initialize a variable, it will default to the basic primitive (`NO`, for booleans). If you override a method you don't need, but need a return value, you usually return `NO`, unless the design or architecture specifies otherwise.

Reply:
For these methods that delivers no value to this class, I'd usually put them at the end of the file, with a prama mark - Unused Methods (or whatever you want). This can save a future fellow mobile engineer time, scrolling through needless methods, just to get to the meat (getNativeAdWithAdTypes:xxx). `networkExtrasClass`, and `networkExtrasClass` are cool though, since they actually serve some purpose for this class.

Reply:
The default value for [`isBannerAnimationOK`](https://developers.google.com/mobile-ads-sdk/docs/dfp/ios/api/reference/Protocols/GADMAdNetworkAdapter#-isbanneranimationok) is `YES`. So leaving as is.